ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Cleaner.java

```diff
@@ -78,21 +77,22 @@
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicBoolean;
-import java.util.concurrent.atomic.AtomicInteger;
 import java.util.stream.Collectors;
 
 import static org.apache.hadoop.hive.conf.Constants.COMPACTOR_CLEANER_THREAD_NAME_FORMAT;
+import static org.apache.hadoop.hive.conf.HiveConf.ConfVars.HIVE_COMPACTOR_CLEANER_MAX_RETRY_ATTEMPTS;
 import static org.apache.hadoop.hive.conf.HiveConf.ConfVars.HIVE_COMPACTOR_CLEANER_RETENTION_TIME;
 import static org.apache.hadoop.hive.conf.HiveConf.ConfVars.HIVE_COMPACTOR_DELAYED_CLEANUP_ENABLED;
 import static org.apache.hadoop.hive.metastore.HMSHandler.getMSForConf;
 import static org.apache.hadoop.hive.metastore.utils.MetaStoreUtils.getDefaultCatalog;
 
-import com.codahale.metrics.Counter;
-
 /**
  * A class to clean directories after compactions.  This will run in a separate thread.
  */
 public class Cleaner extends MetaStoreCompactorThread {
+
+  static final String CURRENT_CLEANER_RETRY_ATTEMPTS = "hive.compactor.cleaner.retry.currentattempts";
```

Comment:
why not use HiveConf var?

Reply:
The value for this is not a global configuration. This setting holds the _actual_  retry attempt count _per compaction setting_, therefore I think it is not necessary to have a hive config for this. Any value set would be ignored anyways because of the above.

Reply:
I didn't say we need to create a new one, rather reuse an existing one, however, as I can see config defined in HiveConf is used to set the max attempts. closing this